Python/dynload_aix.c

```diff
@@ -144,10 +144,18 @@ aix_loaderror(const char *pathname)
         ERRBUF_APPEND(message[i]);
         ERRBUF_APPEND("\n");
     }
-    errbuf[strlen(errbuf)-1] = '\0';            /* trim off last newline */
-    pathname_ob = PyUnicode_FromString(pathname);
-    errbuf_ob = PyUnicode_FromString(errbuf);
-    PyErr_SetImportError(errbuf_ob, NULL, pathname);
+    /* Subtract 1 from the length to trim off trailing newline */
+    errbuf_ob = PyUnicode_DecodeLocaleAndSize(errbuf, strlen(errbuf)-1, "surrogateescape");
+    if (errbuf_ob == NULL) {
+        return;
+    }
+    pathname_ob = PyUnicode_DecodeLocale(pathname, "surrogateescape");
```

Comment:
For `pathname` we use `PyUnicode_EncodeFSDefault()` in `_PyImport_LoadDynamicModuleWithSpec()`, so I think we should use `PyUnicode_DecodeFSDefault()` here.

And please use the FS encoding  for pathname and locale encoding for system error message also in `dynload_shlib.c` and `/dynload_hpux.c`.

Reply:
Ok, so error messages use PyUnicode_DecodeLocale w/ surrogateescape but for filesystem paths use PyUnicode_DecodeFSDefault. What about shortname, eg. https://github.com/python/cpython/blob/master/Python/dynload_shlib.c#L112 ?

Reply:
For dynload_hpux, it seems that the error message is composed of an ASCII-only literal string and the path: https://github.com/python/cpython/blob/master/Python/dynload_hpux.c#L37-L39 so it would seem that PyUnicode_DecodeFSDefault would be best here, no?

Reply:
Answering my own question about encoding of shortname. According to PEP 489, it will always be encoded in ASCII: https://www.python.org/dev/peps/pep-0489/#export-hook-name - either directly or by being encoded with "punycode".